crypto/s2n_ecc_evp.h

```diff
@@ -36,7 +36,7 @@ extern const struct s2n_ecc_named_curve s2n_ecc_curve_secp256r1;
 extern const struct s2n_ecc_named_curve s2n_ecc_curve_secp384r1;
 
 #define S2N_ECC_EVP_SUPPORTED_CURVES_COUNT 2
-#if S2N_OPENSSL_VERSION_AT_LEAST(1, 1, 0) && !defined(LIBRESSL_VERSION_NUMBER)
+#if S2N_OPENSSL_VERSION_AT_LEAST(1, 1, 0) && !defined(LIBRESSL_VERSION_NUMBER) && !defined(OPENSSL_IS_BORINGSSL)
```

Comment:
Currently BoringSSL/AWS-LC only supports Curve` x25519` curves due to this usage of this API `EVP_PKEY_get1_tls_encodedpoint` in versions 1.1.0 and higher as mentioned in this comment https://github.com/awslabs/aws-lc/issues/63#issuecomment-594186359. Maybe we can add a comment her for future reference.